server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -417,25 +454,26 @@ public void onFailure(Exception e) {
             // are on the cluster state thread and our custom future implementation will throw an
             // assertion.
             if (latch.await(10, TimeUnit.SECONDS) == false) {
-                logger.warn("failed to update remote cluster connection [{}] within {}", clusterAlias, TimeValue.timeValueSeconds(10));
+                logger.warn(
+                    "project [{}] failed to update remote cluster connection [{}] within {}",
+                    projectId,
+                    clusterAlias,
+                    TimeValue.timeValueSeconds(10)
+                );
             }
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
         }
     }
 
-    /**
-     * This method updates the list of remote clusters. It's intended to be used as an update consumer on the settings infrastructure
-     *
-     * @param clusterAlias a cluster alias to discovery node mapping representing the remote clusters seeds nodes
-     * @param newSettings the updated settings for the remote connection
-     * @param listener a listener invoked once every configured cluster has been connected to
-     */
+    // Package-access for testing.
+    @FixForMultiProject(description = "Refactor to supply the project ID associated with the alias and settings, or eliminate this method.")
     void updateRemoteCluster(String clusterAlias, Settings newSettings, ActionListener<RemoteClusterConnectionStatus> listener) {
```

Comment:
IIUC, this method only has test usages after this PR. So yeah I'd prefer to remove it eventually.